src/js/Content/Features/Common/FSkipAgecheck.js

```diff
@@ -1,32 +1,44 @@
 import {SyncedStorage} from "../../../Core/Storage/SyncedStorage";
-import {CallbackFeature} from "../../Modules/Feature/CallbackFeature";
+import {ContextType, Feature} from "../../modulesContent";
 
-export default class FSkipAgecheck extends CallbackFeature {
+export default class FSkipAgecheck extends Feature {
 
     checkPrerequisites() {
         return SyncedStorage.get("send_age_info");
     }
 
-    setup() {
-        this.callback();
-    }
+    apply() {
 
-    callback() {
-        const ageYearNode = document.querySelector("#ageYear");
-        if (ageYearNode) {
-            const myYear = Math.floor(Math.random() * 75) + 10;
-            ageYearNode.value = `19${myYear}`;
-            document.querySelector(".btnv6_blue_hoverfade").click();
-        } else {
-            const btn = document.querySelector(".agegate_text_container.btns a");
+        if (this.context.type === ContextType.AGECHECK) {
+
+            const ageYearNode = document.querySelector("#ageYear");
+            if (ageYearNode) {
+                const myYear = Math.floor(Math.random() * 75) + 10;
+                ageYearNode.value = `19${myYear}`;
+            }
+
+            const btn = document.querySelector(".agegate_text_container.btns > a");
             if (btn && btn.getAttribute("href") === "#") {
                 btn.click();
             }
+        } else {
+            this._skipContentWarning();
+
+            // display changes to none if content warning is shown
+            const node = document.querySelector(".responsive_page_template_content");
+            if (node && node.style.display !== "block") {
+                new MutationObserver((mutations, observer) => {
+                    observer.disconnect();
+                    this._skipContentWarning();
+                }).observe(node, {"attributes": true});
+            }
```

Comment:
This is moved from CApp, with the observed node changed according to Steam's script. The old `.apphub_background` was used for "non-responsive" pages, which used to be the case for sharedfiles pages, but has since been updated. I can't find any "non-responsive" pages in steamcommunity anymore.

I'm not sure why this observer is needed though. Is it in case our script runs before Steam displays the content warning?

Reply:
Yes, it is there for this case